### DISCUSSION TITLE
Search with user id is not woking?

### DISCUSSION THREAD

**Discussion by arunvenkat1911** [mem0ai/mem0#1997] [Q&A]
Hi, 

Search method is not giving me the saved data when re run the code. But the data is there on db file. 

related_memories = m.search(query="where is Alice located?", user_id="alice")

Please assist. 

But this search with memory id is working.
history = m.history(memory_id="61affbb1-13a2-40d5-ae39-2d2e2da7d1d5")

Below is the full code:

**from dotenv import load_dotenv
import os
import sqlite3

load_dotenv()

from mem0 import Memory

history_db_path = "history.db"  # Ensure the path is correct

# Configuration
config = {
    "llm": {
        "provider": "openai",
        "config": {
            "model": "gpt-4o-mini",
            "temperature": 0.2,
            "max_tokens": 1500,
        }
    },
    "history_db_path": history_db_path
}

m = Memory.from_config(config)

# Search memory
related_memories = m.search(query="where is Alice located?", user_id="alice")**

Arun.

**Comment by Dev-Khant** [comment]
Hey @arunvenkat1911, could you please give it another try? We've updated the logic, and several versions have been released since then.
